contracts/libraries/PoolSwapLibrary.sol

```diff
@@ -247,43 +270,91 @@ library PoolSwapLibrary {
+            );
+            priceChangeData.shortBalance = ABDKMathQuad.toUInt(
+                ABDKMathQuad.mul(
+                    ABDKMathQuad.fromUInt(priceChangeData.shortBalance),
+                    ABDKMathQuad.sub(ONE, poolMultiplier)
+                )
+            );
+        } else if (newPrice < oldPrice && priceChangeData.longBalance > 0) {
+            // Price decreased
+            // Using the sigmoid function defined in the function's natspec, move funds from long side to short side
+            bytes16 ratio = divInt(priceChangeData.newPrice, priceChangeData.oldPrice);
+            bytes16 poolMultiplier = sigmoid(leverageAmount, ratio);
+
+            priceChangeData.shortBalance += ABDKMathQuad.toUInt(
+                ABDKMathQuad.mul(ABDKMathQuad.fromUInt(priceChangeData.longBalance), poolMultiplier)
+            );
+            priceChangeData.longBalance = ABDKMathQuad.toUInt(
+                ABDKMathQuad.mul(
+                    ABDKMathQuad.fromUInt(priceChangeData.longBalance),
+                    ABDKMathQuad.sub(ONE, poolMultiplier)
+                )
+            );
+        }
+
+        if (sumBeforePriceChange > priceChangeData.longBalance + priceChangeData.shortBalance) {
+            // Move dust into winning side
+            // This is only ever 1 wei (negligible)
+            if (newPrice > oldPrice) {
+                priceChangeData.longBalance +=
+                    sumBeforePriceChange -
+                    (priceChangeData.longBalance + priceChangeData.shortBalance);
+            } else {
+                priceChangeData.shortBalance +=
+                    sumBeforePriceChange -
+                    (priceChangeData.longBalance + priceChangeData.shortBalance);
+            }
         }
 
-        return (longBalance, shortBalance, longFeeAmount, shortFeeAmount);
+        return (priceChangeData.longBalance, priceChangeData.shortBalance, longFeeAmount, shortFeeAmount);
+    }
+
+    /**
+     * @notice Use a sigmoid function to determine the losing pool multiplier.
+     * @return The losing pool multiplier, represented as an ABDKMathQuad IEEE754 quadruple-precision binary floating-point numbers
+     * @dev The returned value is used in `calculateValueTransfer` as the portion to move from the losing side into the winning side
+     */
+    function sigmoid(bytes16 leverage, bytes16 ratio) private pure returns (bytes16) {
+        /**
+         * denominator = 1 + e ^ (-2 * leverage * (1 - ratio))
+         */
+        bytes16 denominator = ABDKMathQuad.mul(ABDKMathQuad.fromInt(-2), leverage);
+        denominator = ABDKMathQuad.mul(denominator, ABDKMathQuad.sub(ONE, ratio));
+        denominator = ABDKMathQuad.add(ONE, ABDKMathQuad.exp(denominator));
+        bytes16 numerator = ABDKMathQuad.add(ONE, ONE); // 2
+        return ABDKMathQuad.sub((ABDKMathQuad.div(numerator, denominator)), ONE);
```

Comment:
do we need a div by zero check or is `e ^ (-2 * leverage * (1 - ratio))` immune to being `-1`?

Reply:
`x > 0 -> x^y > 0` so I think it's OK.